x/group/server/testsuite/suite.go

```diff
@@ -7,7 +7,7 @@ import (
 	"strings"
 	"time"
 
-	"github.com/regen-network/regen-ledger/testutil"
+	"github.com/regen-network/regen-ledger/types/testutil"
```

Comment:
Are we sure testutil should go under types?

testutil imports `github.com/regen-network/regen-ledger/app` to run the test suites with our app. But then we would have `types -> app -> types` import cycle.

Reply:
I am not sure. @aaronc mentioned [here](https://github.com/regen-network/regen-ledger/issues/319#issue-852548887) perhaps it is a good idea

Reply:
Breaking the cyclic dependency on `app` is exactly what we want to do. It is problematic that module test suites depend on app which depends on modules. Instead, we want to find a way to bootstrap a whole app in each module's test suite. That is one of the key hard pieces of this refactoring. So how can we move `testutil` -> `types/testutil` and remove the `app` dependency?

Reply:
A simple alternative way to start would be to not create a separate `app/` go module to start and just create modules for `x/*` modules and `types/`. Then we move the relevant stuff from `testutil` that `x/*` modules need into `types/` without importing `app/`. Which means all of the cli test suites will take a `network.Config` parameter but not actually run tests (as in https://github.com/cosmos/cosmos-sdk/pull/6711). Then in `app/` we will actually run those test suites passing in a config which uses our `RegenApp`. If that's not clear it might be useful to get on a call to chat about it. @amaurym does this approach make sense to you?

Reply:
Just jumped out of a call with Andrei discussing this. Yeah this makes sense, and we agreed this solves cyclic dependency issues. Just to double-check: `x/*` won't have any `_test.go` files anymore, they just define test suites, which `app/` imports and runs, is that right?

Reply:
> Just jumped out of a call with Andrei discussing this. Yeah this makes sense, and we agreed this solves cyclic dependency issues. Just to double-check: `x/*` won't have any `_test.go` files anymore, they just define test suites, which `app/` imports and runs, is that right?

It will have some `_test.go` files I think but just not ones that depend on `app/`.